lxd/daemon.go

```diff
@@ -2435,6 +2435,55 @@ func (d *Daemon) heartbeatHandler(w http.ResponseWriter, r *http.Request, isLead
 
 		logger.Info("Partial heartbeat received", logger.Ctx{"local": localClusterAddress})
 	}
+
+	// Refresh cluster member resource info cache.
+	var muRefresh sync.Mutex
+
+	for _, member := range hbData.Members {
+		// Ignore offline servers.
+		if !member.Online {
+			continue
+		}
+
+		if member.Name == s.ServerName {
+			continue
+		}
+
+		go func(name string, address string) {
+			muRefresh.Lock()
```

Comment:
I'm quite confused by this lock.
We seem to be starting a go routine per online cluster member (so presumably there is some intention for concurrency), and then we serialise the whole function it via a mutex.

Is the intention to launch a cluster CPU flags refresh task and not hold up the rest of the `heartbeatHandler` function? 

If so, we could move this logic into its own function (as suggested above) and then we can run it via `go newfunction()` and then have the actual refresh loop done sequentially.